src/components/Card.js

```diff
@@ -49,17 +89,44 @@ export class Card {
     this._deleteButton = this._cardElement.querySelector(
       ".card__delete-button"
     );
-    this._deleteButton.addEventListener("click", () => {
-      this._deleteCard();
-    });
+    this._deleteButton.addEventListener("click", this._handleDelete);
   }
 
-  _deleteCard() {
+  deleteCard() {
+    console.log("you pressed this delete button");
+    // this._handleDelete.setEventListeners();
+    // this._handleDelete.open();
     this._cardElement.remove();
     this._cardElement = null;
   }
 
+  refreshLikesCount(likesCount) {
+    likesCount < 1
+      ? (this._cardElement.querySelector(".card__likes-counter").textContent =
+          "")
+      : (this._cardElement.querySelector(".card__likes-counter").textContent =
+          likesCount);
+  }
+
   _handleLike() {
-    this._likeButton.classList.toggle("card__like-button_active");
+    if (this._isLiked) {
+      this._handleDislikeCallback();
+      this.dislike();
```

Comment:
` this.dislike();`

You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.

Reply:
Thanks for the review. I am not very clear what should I do in this one, since the code all does is check the cards I liked before and add the hearth-active class in whichever card I liked before. 

How should I proceed?

Reply:
> How should I proceed?

You've done it correly now. You do in in `index.js` in blocks `then`. So, everything is ok